soccer/main.py

```diff
@@ -14,16 +15,48 @@
 LEAGUE_IDS = leagueids.LEAGUE_IDS
 TEAM_NAMES = teamnames.team_names
 
+
+def get_input_key():
+    """Input API key and validate"""
+    print("No API key found!")
+    print("Please visit {0} and get an API token.".format(BASE_URL))
+    while True:
+        confkey = raw_input("Please enter API key: ")
+        if len(confkey) == 32:  # 32 chars
+            try:
+                int(confkey, 16)  # hexadecimal
+            except ValueError:
+                print("Invalid API key, try again")
+            else:
+                break
+        else:
+            print("Invalid API key, try again")
+    return confkey
+
+
+def load_config_key():
+    """Load API key from config file, write if needed"""
+    home = os.path.expanduser("~")
+    config = os.path.join(home, "soccer-cli.ini")
+    if not os.path.exists(config):
+        with open(config, "w") as cfile:
+            key = get_input_key()
+            cfile.write(key)
+    else:
+        with open(config, "r") as cfile:
+            key = cfile.read()
+    return key
+
+
 try:
     api_token = os.environ['SOCCER_CLI_API_TOKEN']
 except KeyError:
-    from soccer.config import config
-    api_token = config.get('SOCCER_CLI_API_TOKEN')
+    api_token = load_config_key()
```

Comment:
I agree. My comment was geared more towards removing it from the:

```python
   try:
      api_token = os.environ['SOCCER_CLI_API_TOKEN']
  except KeyError:
     api_token = load_config_key()
```

and moving this into the `load_config_key()`

```python
   def load_config_key():
        try:
           api_token = os.environ['SOCCER_CLI_API_TOKEN']
        except KeyError:
           # load from user file
          ...

@click.option('--APIKey', prompt=True,
              default=load_config_key)

```